tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,6 +66,699 @@ public function test_wp_enqueue_script() {
+
+		return $data;
+	}
+
+	/**
+	 * Test valid defer loading with async dependent.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_defer_with_async_dependent() {
+		// case with one async dependent.
+		wp_enqueue_script( 'main-script-d4', '/main-script-d4.js', array(), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'dependent-script-d4-1', '/dependent-script-d4-1.js', array( 'main-script-d4' ), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'dependent-script-d4-2', '/dependent-script-d4-2.js', array( 'dependent-script-d4-1' ), null, array( 'strategy' => 'async' ) );
+		wp_enqueue_script( 'dependent-script-d4-3', '/dependent-script-d4-3.js', array( 'dependent-script-d4-2' ), null, array( 'strategy' => 'defer' ) );
+		$output    = get_echo( 'wp_print_scripts' );
+		$expected  = "<script type='text/javascript' src='/main-script-d4.js' id='main-script-d4-js' defer></script>\n";
+		$expected .= "<script type='text/javascript' src='/dependent-script-d4-1.js' id='dependent-script-d4-1-js' defer></script>\n";
+		$expected .= "<script type='text/javascript' src='/dependent-script-d4-2.js' id='dependent-script-d4-2-js' defer></script>\n";
+		$expected .= "<script type='text/javascript' src='/dependent-script-d4-3.js' id='dependent-script-d4-3-js' defer></script>\n";
+
+		$this->assertSame( $expected, $output );
+	}
+
+	/**
+	 * Test invalid defer loading strategy case.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_loading_strategy_with_invalid_defer_registration() {
+		// Main script is defer and all dependent are not defer. Then main script will have blocking(or no) strategy.
+		wp_enqueue_script( 'main-script-d4', '/main-script-d4.js', array(), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'dependent-script-d4-1', '/dependent-script-d4-1.js', array( 'main-script-d4' ), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'dependent-script-d4-2', '/dependent-script-d4-2.js', array( 'dependent-script-d4-1' ), null, array( 'strategy' => 'blocking' ) );
+		wp_enqueue_script( 'dependent-script-d4-3', '/dependent-script-d4-3.js', array( 'dependent-script-d4-2' ), null, array( 'strategy' => 'defer' ) );
+		$output   = get_echo( 'wp_print_scripts' );
+		$expected = "<script type='text/javascript' src='/main-script-d4.js' id='main-script-d4-js'></script>\n";
+		$this->assertStringContainsString( $expected, $output );
+	}
+
+	/**
+	 * Test valid blocking loading strategy cases.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_loading_strategy_with_valid_blocking_registration() {
+		wp_enqueue_script( 'main-script-b1', '/main-script-b1.js', array(), null, array( 'strategy' => 'blocking' ) );
+		$output   = get_echo( 'wp_print_scripts' );
+		$expected = "<script type='text/javascript' src='/main-script-b1.js' id='main-script-b1-js'></script>\n";
+		$this->assertSame( $expected, $output );
```

Comment:
Please add message.

Reply:
Context:
> All PHPUnit assertions, as well as all WordPress custom assertions, allow for a `$message` parameter to be passed. This message will be displayed when the assertion fails and can help immensely when debugging a test. **This parameter should always be used if more than one assertion is used in a test method.**
> [Ref](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions)

(my emphasis)

Reply:
Thanks @costdev @spacedmonkey I've addressed these in the same PR that I addressed `@covers` in, please see https://github.com/10up/wordpress-develop/pull/61

I'll add you both as reviewers against that PR.

Reply:
Seems like I can not add either of you as reviewers, but I'd appreciate the eyes and opinions on this.